dbms/src/Flash/Mpp/MPPTaskId.h

```diff
@@ -55,6 +55,27 @@ struct MPPQueryIdHash
     size_t operator()(MPPQueryId const & mpp_query_id) const noexcept;
 };
 
+/// A MPP query has one or more MPPGathers, each mpp gather has one or more MPPTasks. The mpp tasks in different mpp gathers are independent
+/// to each other, so theoretically speaking, the smallest cancel/retry unit of MPP query could be MPPGather. However, MPPGathers belong to
+/// the same MPP query could have dependence to each other(e.g. for query A join B, if the join is not supported in TiFlash, TiDB will generate
+/// two mpp gathers, one is reading from A and the other is reading from B, the probe side's mpp gather depends on the build side's mpp gather),
+/// so the smallest scheduling unit in TiFlash is still the MPP query
+struct MPPGatherId
```

Comment:
I'm a little confused, isn't the current MPPQueryId at the gather level?

Reply:
As mentioned in the case, if there are gather1 join gather2, will gather2 be cancelled when gather1 is cancelled due to limit? Because the cancellation is based on MPPQueryId.

Reply:
Currently, the cancel unit is MPPQuery, but we can implement mpp gather level's cancel. In this pr, all the `gather_id` is hard coded as 0, which mean even for `A join B`, TiFlash still get only one MPPGather

Reply:
Add more comments.

Reply:
ok